test/functional/tests/io_class/test_io_class_stats_file_size_core.py

```diff
@@ -0,0 +1,233 @@
+#
+# Copyright(c) 2022 Intel Corporation
+# Copyright(c) 2024 Huawei Technologies Co., Ltd.
+# SPDX-License-Identifier: BSD-3-Clause
+#
+
+import re
+import pytest
+
+from datetime import timedelta
+from api.cas import casadm, ioclass_config
+from api.cas.cache_config import CacheMode, CleaningPolicy, SeqCutOffPolicy
+from api.cas.ioclass_config import IoClass
+from core.test_run_utils import TestRun
+from storage_devices.disk import DiskTypeSet, DiskType, DiskTypeLowerThan
+from test_tools import fs_utils
+from test_tools.disk_utils import Filesystem
+from test_tools.fio.fio import Fio
+from test_tools.fio.fio_param import IoEngine, ReadWrite
+from test_utils.os_utils import sync, drop_caches
+from test_utils.size import Size, Unit
+from tests.io_class.io_class_common import template_config_path
+
+
+@pytest.mark.parametrizex("cache_mode", CacheMode)
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+def test_io_class_stats_file_size_core_fs(cache_mode: CacheMode):
+    """
+    title: Open CAS statistics values per core for IO classification by file size.
+    description: |
+      Check Open CAS ability to assign correct IO class based on the file size.
+      Test checking configuration with different filesystems on each core.
+    pass_criteria:
+        - after FIO with direct statistics increase only for direct IO class
+    """
+
+    with TestRun.step("Prepare devices"):
+        cache_device = TestRun.disks['cache']
+        cache_device.create_partitions([Size(10, Unit.GibiByte)])
+        cache_device = cache_device.partitions[0]
+
+        core_device = TestRun.disks['core']
+        core_device.create_partitions([Size(5, Unit.GibiByte)] * len(Filesystem))
```

Comment:
Instead of creating core per each filesystem, maybe add single core and parametrize test with filesystem?